core/domain/subtopic_page_domain.py

```diff
@@ -92,25 +95,117 @@ def to_dict(self):
+        self.content_ids_to_audio_translations = (
+            content_ids_to_audio_translations)
+
+    def validate(self):
+        self.subtitled_html.validate()
+
+        # TODO(tjiang11): Extract content ids to audio translations out into
+        # its own object to reuse throughout audio-capable structures.
+        if not isinstance(self.content_ids_to_audio_translations, dict):
+            raise utils.ValidationError(
+                'Expected content_ids_to_audio_translations to be a dict,'
+                'received %s' % self.content_ids_to_audio_translations)
+        for (content_id, audio_translations) in (
+                self.content_ids_to_audio_translations.iteritems()):
+
+            if not isinstance(content_id, basestring):
+                raise utils.ValidationError(
+                    'Expected content_id to be a string, received: %s' %
+                    content_id)
+            if not isinstance(audio_translations, dict):
+                raise utils.ValidationError(
+                    'Expected audio_translations to be a dict, received %s'
+                    % audio_translations)
+
+            allowed_audio_language_codes = [
+                language['id'] for language in (
+                    constants.SUPPORTED_AUDIO_LANGUAGES)]
+            for language_code, translation in audio_translations.iteritems():
+                if not isinstance(language_code, basestring):
+                    raise utils.ValidationError(
+                        'Expected language code to be a string, received: %s' %
+                        language_code)
+
+                if language_code not in allowed_audio_language_codes:
+                    raise utils.ValidationError(
+                        'Unrecognized language code: %s' % language_code)
+
+                translation.validate()
+
+        content_ids = set([self.subtitled_html.content_id])
+
+        audio_content_ids = set(
+            [audio[0] for audio
+             in self.content_ids_to_audio_translations.iteritems()])
+        if content_ids != audio_content_ids:
+            raise utils.ValidationError(
+                'Expected content_ids_to_audio_translations to contain '
+                'all content_ids in the subtopic page. Expected: %s\n '
+                'Actual: %s' % (content_ids, audio_content_ids))
+
+    @classmethod
+    def create_default_subtopic_page_contents(cls):
+        return cls(
+            state_domain.SubtitledHtml.create_default_subtitled_html(
+                'content'), {'content': {}})
+
+    def to_dict(self):
+        """Returns a dict representing this SubtopicPageContents domain object.
+
+        Returns:
+            A dict, mapping all fields of SubtopicPageContents instance.
+        """
+        return {
+            'subtitled_html': self.subtitled_html.to_dict(),
+            'content_ids_to_audio_translations': (
+                self.content_ids_to_audio_translations),
```

Comment:
`content_ids_to_audio_translations` is going to have "AudioTranslation" object within it so we will have to convert it to a dict. right?

Reply:
done